compiler/README.md

```diff
@@ -52,70 +54,118 @@ protobuf {
 }
 ```
 
+- #### Maven
+
 For Maven, include the [protobuf plugin](https://www.xolstice.org/protobuf-maven-plugin/) and configure it to use the
-`protoc-gen-grpc-kotlin` compiler, like:
-```
-<plugin>
-  <groupId>org.xolstice.maven.plugins</groupId>
-  <artifactId>protobuf-maven-plugin</artifactId>
-  <version>0.6.1</version>
-  <executions>
-    <execution>
-      <id>compile</id>
-      <goals>
-        <goal>compile</goal>
-        <goal>compile-custom</goal>
-      </goals>
-      <configuration>
-        <protocArtifact>com.google.protobuf:protoc:YOUR_PROTOBUF_VERSION:exe:${os.detected.classifier}</protocArtifact>
-        <pluginId>grpc-java</pluginId>
-        <pluginArtifact>io.grpc:protoc-gen-grpc-java:YOUR_GRPC_VERSION:exe:${os.detected.classifier}</pluginArtifact>
-        <protocPlugins>
-          <protocPlugin>
-            <id>grpc-kotlin</id>
-            <groupId>io.grpc</groupId>
-            <artifactId>protoc-gen-grpc-kotlin</artifactId>
-            <version>YOUR_GRPC_KOTLIN_VERSION</version>
-            <classifier>jdk7</classifier>
-            <mainClass>io.grpc.kotlin.generator.GeneratorRunner</mainClass>
-          </protocPlugin>
-        </protocPlugins>
-      </configuration>
-    </execution>
-    <execution>
-      <id>compile-kt</id>
-      <goals>
-        <goal>compile-custom</goal>
-      </goals>
-      <configuration>
-        <protocArtifact>com.google.protobuf:protoc:YOUR_PROTOBUF_VERSION:exe:${os.detected.classifier}</protocArtifact>
-        <outputDirectory>${project.build.directory}/generated-sources/protobuf/kotlin</outputDirectory>
-        <pluginId>kotlin</pluginId>
-      </configuration>
-    </execution>
-  </executions>
-</plugin>
-```
-
-Make sure you include a dependency on `stub` and protobuf libraries like:
+`protoc-gen-grpc-kotlin` compiler, starting by adding `properties` for the versions of the plugins
+
+```
+<properties>
+    <!-- This is just an example, don't mind the versions -->
+    <java.version></java.version> <!-- you should already have this line -->
+    <kotlin.version>1.5.31</kotlin.version> <!-- you should already have this line -->
+
+    <!- the version is the lower of the ones found in these two links
+    (they should always be synced, but double checking is better than being stuck on an error):
+    https://search.maven.org/search?q=a:protoc-gen-grpc-kotlin
+    https://search.maven.org/search?q=a:grpc-kotlin-stub -->
+    <grpc.kotlin.version>1.2.0</grpc.kotlin.version>
+
+    <!-- The version is the latest found here: https://search.maven.org/artifact/io.grpc/grpc-protobuf/1.42.0/jar -->
+    <java.grpc.version>1.42.0</java.grpc.version>
```

Comment:
I think the max we currently support is `1.39.0`

Reply:
Is there a place where there's track of the maximum supported version for this and `protobuf-kotlin` ?

Reply:
I don't know of a good resource for that currently.  Ultimately we need to get gRPC Kotlin upgraded to work with the latest versions: https://github.com/grpc/grpc-kotlin/issues/289

Reply:
Should be done, would you prefer me to resolve for this and the following points